src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -8518,20 +8518,30 @@ public static ulong Splice(ulong[] first, ulong[] second, ulong[] maskArray, int
         }
 
 
-        private static TElem GetLoadVectorExpectedResultByIndex<TMem, TElem>(int index, TMem[] firstOp, TElem[] result)
+        private static TElem GetLoadVectorExpectedResultByIndex<TMem, TElem>(int index, TElem[] mask, TMem[] data, TElem[] result)
             where TMem  : INumberBase<TMem>
             where TElem : INumberBase<TElem>
         {
-            return (firstOp[index] == TMem.Zero) ? TElem.Zero : TElem.CreateTruncating(firstOp[index]);
+            return (mask[index] == TElem.Zero) ? TElem.Zero : TElem.CreateTruncating(data[index]);
         }
 
-        private static bool CheckLoadVectorBehaviorCore<TMem, TElem>(TMem[] firstOp, TElem[] result, Func<int, TElem, TElem> map)
+        private static TElem GetLoadVectorExpectedResultByIndex<TMem, TElem>(int index, TMem[] data, TElem[] result)
             where TMem  : INumberBase<TMem>
             where TElem : INumberBase<TElem>
         {
-            for (var i = 0; i < firstOp.Length; i++)
+            TElem[] mask = new TElem[result.Length];
+            Array.Fill(mask, TElem.One);
+
+            return GetLoadVectorExpectedResultByIndex(index, mask, data, result);
```

Comment:
>GetLoadVectorExpectedResultByIndex [](http://example.com/codeflow?start=19&length=34)

The original test for this validation is little convulated and hence a bit hard to follow. But here, we are filling all mask and then calling a method that checks for `mask[index] == TElem.Zero`...wondering why can't we just skip calling `GetLoadVectorExpectedResultByIndex` and just do `return TElem.CreateTruncating(data[index])` here?